File: pkg/clientgen/testdata/tsapp/input_httpstatus.ts
-- encore.app --
{"id": ""}

-- package.json --
{"name": "ts-test-app"}

-- svc/svc.ts --
import { api, HttpStatus } from "encore.dev/api";

export const dummy = api(
  { expose: true, method: "GET", path: "/dummy" },
  async (): Promise<Response> => {},
);


interface Response {
  message: string,
  status: HttpStatus,
}

File: pkg/clientgen/testdata/goapp/input_httpstatus.go
-- go.mod --
module app

-- encore.app --
{"id": ""}

-- svc/svc.go --
package svc

type Response struct {
    Message string
    Status int `encore:"httpstatus"`
}

-- svc/api.go --
package svc

import (
    "context"
    "net/http"
)

// DummyAPI is a dummy endpoint.
//encore:api public
func DummyAPI(ctx context.Context) (*Response, error) {
    return nil, nil
}
